.github/workflows/ci.yml

```
@@ -2,34 +2,58 @@ name: CI
 
 on:
   push:
-  pull_request_target:
+  pull_request:
   schedule:
     - cron: "6 20 * * 6"
 
 jobs:
   test:
     runs-on: ubuntu-latest
-    container: ${{matrix.image}}
     strategy:
       fail-fast: false
       matrix:
-        image:
-          - ruby:2.5.0
-          - ruby:2.5
-          - ruby:2.6.0
-          - ruby:2.6
-          - ruby:2.7.0
-          - ruby:2.7
-          - jruby:9.2
-        include:
-          - image: ruby:rc
-            continue-on-error: true
+        ruby:
+          - 2.5.0
+          - 2.5
+          - 2.6.0
+          - 2.6
+          - 2.7.0
+          - 2.7
+          - jruby-9.2
```

```suggestion
          - jruby-9.2
        include:
          - ruby: ruby-head
            continue-on-error: true
          - ruby: jruby-head
            continue-on-error: true
```

This way we don't need `test-rc` section.

---

That doesn't work. See the last commit I pushed.

---

I'm not quite sure what you mean? How can you tell it doesn't work?

---

It was still failing. `continue-on-error` can only be set on a step, not on a job. I worked around that with an interpolation.

---

I was under impression that this would be set on a job (in our case `test`), not a step. I got that from [the documentation](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontinue-on-error). But I'm not quite sure how it interacts with the matrix.

---

Well, I pushed another change with that, and unless I'm overlooking something it doesn't work.

---

I'm getting confused. What does `continue-on-error` do? One of the previous runs just marked "head" build green even though it failed.

That's one way to interpret it but it hides the failure. We already have set up a couple build that must be green in order for PR to be accepted (see "Required" label in the checks list). I'd still want to know when particular builds fail.

Maybe we don't need this `continue-on-error` at all? I certainly don't want to dig through all build to find which are failing. I'd rather see them red in the checks list.

---

> What does continue-on-error do? One of the previous runs just marked "head" build green even though it failed.

Yes that's exactly what it does, and that is why it's a very bad replacement for Travis's `allow_failures` feature. It show the step as green, so you might break the build and nobody will ever notice.

> Maybe we don't need this continue-on-error at all? I certainly don't want to dig through all build to find which are failing. I'd rather see them red in the checks list.

That is exactly what I was initially suggesting.

---

All right. Sorry for the confusion.

I must've mixed it up `fail-fast`.

Let's let particular checks fail but run all of them regardless if any one fails.

---

Done. I also restored the `pull_request_target`, should be good to merge now.